docs/src/DeveloperDocumentation/way_of_working.md

```diff
@@ -0,0 +1,135 @@
+   can be merged faster.
+3. Let us know what you are working on:
+   - You can open a draft pull request on GitHub right at the beginning of your
+     work.
+   - Feel free to contact us on
+     [Slack](https://join.slack.com/t/oscar-system/shared_invite/zt-thtcv97k-2678bKQ~RpR~5gZszDcISw).
+   - Have a look at [our community page](https://oscar.computeralgebra.de/community/).
+4. Please also read our page on [Documenting OSCAR code](@ref).
+5. Look at existing code that does similar things to your project to get an
+   idea of what OSCAR code should look like. Try to look at multiple examples.
+
+## Overview
+In general you have to do
+the following six steps for submitting changes to the Oscar source:
+
+1. Fork the main Oscar.jl repository. For this go to the [Oscar.jl GitHub
+   page](https://github.com/oscar-system/Oscar.jl) and click on "Fork" at the
+   upper right.
+2. Clone your forked repository to your local machine.
+   ```
+   git clone git@github.com:your_github_username/Oscar.jl
+   ```
+3. Create a new branch, usually the naming convention is to use your initials
+   ("yi") and then describe your change, for example:
+   ```
+   git checkout -b yi/new_feature
+   git checkout -b yi/issue1234
+   git checkout -b yi/document_feature
+   ```
+4. Edit your source and try out your changes locally (see below). To use your local copy of
+   the sources, start Julia and
+   ```
+   ]dev /path/to/local/clone/of/your/fork/of/Oscar.jl
+   ```
+   If this succeeds, you can enter `using Oscar` in Julia and it will use your local
+   copy.
+5. Once you are done editing, push your branch and open a pull request. It is
+   recommended that you open a draft pull request to the main Oscar repository
+   as soon as you start working. That way Oscar developers are aware of work
+   being done and can give feedback early in the process.
+6. Once you have finished your work, mark your pull request as ready. It will
+   then be reviewed and, probably after feedback and requests for changes,
+   merged.
+
+## Alternative: `]dev Oscar`
+
+Alternatively you can call
+```
+]dev Oscar
+```
+in Julia. This will create a directory `~/.julia/dev/Oscar`. This directory is
+a git clone of the central Oscar repository. You can develop your code here,
+however you will still have to fork Oscar, as you have no rights to push to the
+central repository. You can then add your fork as another remote, have a look
+at the section on rebasing below for hints.
+
+
+## The edit process
+### Editing the source
+The sources can be found in the `src` folder. Please pay attention to the
+folder structure and choose sensibly where to place your code (when fixing a
+bug this is probably a minor question).
+
+### Adding tests
+There are two ways to add tests:
+  - There are combined tests and examples in the docstrings, namely the
+    `jldoctest` blocks. For these have a closer look at [Documenting OSCAR
+    code](@ref).
```

Comment:
Does the ref work when line wrapped this way? (I have no idea, genuine question)

Reply:
I just tried it out and it actually works. :)